crates/integration_test/src/tests/cgroups/network.rs

```diff
@@ -0,0 +1,146 @@
+use std::path::Path;
+
+use anyhow::{Context, Result};
+use pnet::datalink;
+
+use oci_spec::runtime::{
+    LinuxBuilder, LinuxInterfacePriorityBuilder, LinuxNamespace, LinuxNamespaceType,
+    LinuxNetworkBuilder, LinuxResourcesBuilder, Spec, SpecBuilder,
+};
+use test_framework::{test_result, ConditionalTest, TestGroup, TestResult};
+
+use crate::utils::{test_outside_container, test_utils::check_container_created};
+
+fn create_spec(
+    cgroup_name: &str,
+    class_id: u32,
+    prio: u32,
+    if_name: &str,
+    with_net_ns: bool,
+    with_user_ns: bool,
+) -> Result<Spec> {
+    // Get default namespaces and filter them to optional exclude network or user namespaces
+    let default_namespaces: Vec<LinuxNamespace> = oci_spec::runtime::get_default_namespaces()
+        .into_iter()
+        .filter(|ns| match ns.typ() {
+            LinuxNamespaceType::Network => with_net_ns,
+            LinuxNamespaceType::User => with_user_ns,
+            _ => true,
+        })
+        .collect();
+
+    // Create the Linux Spec
+    let linux_spec = LinuxBuilder::default()
+        .cgroups_path(Path::new("/runtime-test").join(cgroup_name))
+        .resources(
+            LinuxResourcesBuilder::default()
+                .network(
+                    LinuxNetworkBuilder::default()
+                        .class_id(class_id)
+                        .priorities(vec![LinuxInterfacePriorityBuilder::default()
+                            .name(if_name)
+                            .priority(prio)
+                            .build()
+                            .context("failed to build network interface priority spec")?])
+                        .build()
+                        .context("failed to build network spec")?,
+                )
+                .build()
+                .context("failed to build resource spec")?,
+        )
+        .namespaces(default_namespaces)
+        .build()
+        .context("failed to build linux spec")?;
+
+    // Create the top level Spec
+    let spec = SpecBuilder::default()
+        .linux(linux_spec)
+        .build()
+        .context("failed to build spec")?;
+
+    Ok(spec)
+}
+
+fn test_network_cgroups() -> TestResult {
+    let cgroup_name = "test_network_cgroups";
+
+    let interfaces = datalink::interfaces();
+
+    // Gets the loopback interface, or defaults to "lo"
+    let lo_if_name = interfaces.get(0).map_or_else(|| "lo", |iface| &iface.name);
+    // Gets the first ethernet interface, or defaults to "eth0"
+    let eth_if_name = interfaces
+        .get(1)
+        .map_or_else(|| "eth0", |iface| &iface.name);
```

Comment:
It is better to check about `interfaces` in `can_run()`?

Reply:
This is what the runtime tools tests are currently doing, they check the interfaces and default them to `lo` and `eth0`. As far as I can tell the tests may still pass if the interfaces aren't found.

Reply:
See here: https://github.com/opencontainers/runtime-tools/blob/1684d131456a6bc99b8e96aa4a99783f21e58d79/validation/linux_cgroups_network/linux_cgroups_network.go#L21

I wish they explained the reasoning for setting a default more in the comment. It seems the rationale here, based on [this commit message](https://github.com/opencontainers/runtime-tools/commit/cf9decfcf37364d67f0d93ec80a0d0d847cef480) is to allow the test to run on systems with different interface names. For example as far as I can tell on every system the first interface is loopback (lo), and then on many systems the second interface may actually be something other than `eth0`. In fact on my system I do not have `eth0`, and the second interface is `wlp0s20f3`, probably because I don't have an ethernet interface on my laptop.

When I run the test with "eth0" explicitly set as the interface on my laptop the test does in fact fail. So I think you're right, we should probably add this to `can_run`.

Reply:
Take a look at the latest revision.